src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +22,26 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String inputLine = "";
+            while ((inputLine = bufferedReader.readLine()) != null) {
+                stringBuilder.append(inputLine.toLowerCase().replaceAll("[^a-z]", ""));
+            }
```

Comment:
you can find a way to iterate over each character already at this point
then, using a method in class Character, you can check if the specific char suits you
and add it to the set

Reply:
I didn't remove a BufferedReader. But changed  from readLine() to read().